### PR TITLE
[Token] Store the framework reserved accounts unders goverance

### PR DESCRIPTION
**Pull request by areshand**
### Description

put 0x2 .. 0x10 under aptos governance

### Test Plan
update unit test

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3365)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/genesis.move [hunk 1]
```diff
@@ -49,6 +49,8 @@ module aptos_framework::genesis {
         // deployed to. This will be entirely managed by on-chain governance and no entities have the key or privileges
         // to use this account.
         let (aptos_framework_account, framework_signer_cap) = account::create_aptos_framework_account();
+        let (aptos_token_account, aptos_token_signer_cap) = account::create_framework_reserved_account(@0x3);
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/genesis.move:52]: can we unify these, no need for two calls.

File: aptos-move/framework/aptos-framework/sources/genesis.move [hunk 2]
```diff
@@ -61,6 +63,7 @@ module aptos_framework::genesis {
 
         // Give the decentralized on-chain governance control over the core framework account.
         aptos_governance::store_signer_cap(&aptos_framework_account, @aptos_framework, framework_signer_cap);
+        aptos_governance::store_signer_cap(&aptos_token_account, @0x3, aptos_token_signer_cap);
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/genesis.move:66]: same

File: aptos-move/framework/aptos-framework/sources/account.move [hunk 3]
```diff
@@ -486,6 +486,13 @@ module aptos_framework::account {
         (signer, signer_cap)
     }
 
+    /// create the account for system reserved addresses
+    public(friend) fun create_framework_reserved_account(addr: address): (signer, SignerCapability) {
+        let signer = create_account_unchecked(addr);
+        let signer_cap = SignerCapability { account: addr };
+        (signer, signer_cap)
+    }
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/account.move:494]: this should make a check that it is between 0x1 and 0x10 or something...

File: aptos-move/framework/aptos-framework/sources/genesis.move (hunk 1 above)
Comment (areshand) [reply aptos-move/framework/aptos-framework/sources/genesis.move]: I remove create_aptos_framework_account and replace it with create_framework_reserved_account

File: aptos-move/framework/aptos-framework/sources/genesis.move (hunk 2 above)
Comment (areshand) [reply aptos-move/framework/aptos-framework/sources/genesis.move:66]: it is more clear to store signer_cap per account. If I have to squash everything in one function call, each parameter is a vector and need to align the accounts in different vector parameters.

File: aptos-move/framework/aptos-framework/sources/account.move (hunk 3 above)
Comment (areshand) [reply aptos-move/framework/aptos-framework/sources/account.move]: updated

File: aptos-move/framework/aptos-framework/sources/account.move [hunk 4]
```diff
@@ -479,10 +481,23 @@ module aptos_framework::account {
...
     }
 
-    /// Create the account for @aptos_framework to help module upgrades on testnet.
-    public(friend) fun create_aptos_framework_account(): (signer, SignerCapability) {
-        let signer = create_account_unchecked(@aptos_framework);
-        let signer_cap = SignerCapability { account: @aptos_framework };
+    /// create the account for system reserved addresses
+    public(friend) fun create_framework_reserved_account(addr: address): (signer, SignerCapability) {
+        assert!(
+            addr == @0x1 ||
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/account.move:487]: Can this just be addr <= @ 0x10 ?

File: aptos-move/framework/aptos-framework/sources/genesis.move [hunk 5]
```diff
@@ -48,7 +48,9 @@ module aptos_framework::genesis {
         // Initialize the aptos framework account. This is the account where system resources and modules will be
         // deployed to. This will be entirely managed by on-chain governance and no entities have the key or privileges
         // to use this account.
-        let (aptos_framework_account, framework_signer_cap) = account::create_aptos_framework_account();
+        let (aptos_framework_account, framework_signer_cap) = account::create_framework_reserved_account(@aptos_framework);
+        let (aptos_token_account, aptos_token_signer_cap) = account::create_framework_reserved_account(@0x3);
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/genesis.move:52]: I thought we wanted 0x2, why 0x3?

File: aptos-move/framework/aptos-framework/sources/account.move (hunk 4 above)
Comment (areshand) [reply aptos-move/framework/aptos-framework/sources/account.move:487]: no, address doesn't support <=

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/account.move:487]: Dang that sucks :(

File: aptos-move/framework/aptos-framework/sources/genesis.move (hunk 5 above)
Comment (areshand) [reply aptos-move/framework/aptos-framework/sources/genesis.move:52]: token is deployed to 0x3. previous discussion was 0x2 is reserved for library in future

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/genesis.move:52]: Okay sounds good

File: aptos-move/framework/aptos-framework/sources/account.move [hunk 6]
```diff
@@ -479,10 +481,23 @@ module aptos_framework::account {
...
+            addr == @0x2 ||
+            addr == @0x3 ||
+            addr == @0x4 ||
+            addr == @0x5 ||
+            addr == @0x6 ||
+            addr == @0x7 ||
+            addr == @0x8 ||
+            addr == @0x9 ||
+            addr == @0x10,
+            ENO_VALID_FRAMEWORK_RESERVED_ADDRESS,
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/account.move:497]: Use error:: to wrap the error code.

File: aptos-move/framework/aptos-framework/sources/genesis.move (hunk 1 above)
Comment (davidiw) [reply aptos-move/framework/aptos-framework/sources/genesis.move]: sorry I meant two different functions not having a single function that returns a bunch of signers :)

**Review by davidiw — APPROVED** [review]
unblocking but looks good after Kevin's concern.

File: aptos-move/framework/aptos-framework/sources/system_addresses.move [hunk 7]
```diff
@@ -29,7 +29,7 @@ module aptos_framework::system_addresses {
     }
 
     public fun is_aptos_framework_address(addr: address): bool {
-        addr == @aptos_framework
+        addr == @aptos_framework || addr == @0x3
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/system_addresses.move:32]: This doesn't feel right to me. Why group these together? Setters in token that need to check the signer should assert against 0x3 distinctly, not 0x1.

Comment (davidiw) [reply aptos-move/framework/aptos-framework/sources/system_addresses.move:32]: right, this should be `is_aptos_token_address` or something

or could be a new function `is_aptos_governance_address`

Comment (areshand) [reply aptos-move/framework/aptos-framework/sources/system_addresses.move]: updated

File: aptos-move/framework/aptos-framework/sources/aptos_governance.move [hunk 8]
```diff
@@ -127,7 +127,7 @@ module aptos_framework::aptos_governance {
         signer_address: address,
         signer_cap: SignerCapability,
     ) acquires GovernanceResponsbility {
-        system_addresses::assert_aptos_framework(aptos_framework);
+        system_addresses::assert_framework_reserved_address(aptos_framework);
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/aptos_governance.move:130]: Why does this need to change? You can still pass the aptos_framework account here right?

**Review by movekevin — APPROVED** [review]

**Pull request merged**